Public/Src/Cache/ContentStore/Distributed/NuCache/CheckpointManager.cs

```diff
@@ -170,11 +152,55 @@ private async Task<bool> CreateCheckpointIncrementalAsync(OperationContext conte
 
             await _checkpointRegistry.RegisterCheckpointAsync(context, checkpointId, sequencePoint).ThrowIfFailure();
             UpdateIncrementalCheckpointInfo(newCheckpointInfo);
-            successfullyUpdatedIncrementalState = true;
-            return successfullyUpdatedIncrementalState;
         }
 
-        private void WriteCheckpointInfo(AbsolutePath path, Dictionary<string, string> newCheckpointInfo)
+        private async Task UploadFilesAsync(OperationContext context, List<AbsolutePath> files, ConcurrentDictionary<string, string> newCheckpointInfo, string incrementalCheckpointsPrefix)

```

Comment:
Can be just IEnumerable<AbsolutePath>, which also means we can avoid ToList conversion above.

Reply:
I actually would slightly prefer to have ToList on the source to separate enumeration process from the consumption process...

---
In reply to: [319297335](https://github.com/microsoft/BuildXL/pull/818#discussion_r319297335) [](ancestors = 319297335)